worker/client.go

```diff
@@ -76,10 +76,15 @@ func (c *Client) RHPFund(ctx context.Context, contractID types.FileContractID, h
 
 // RHPPriceTable fetches a price table for a host.
 func (c *Client) RHPPriceTable(ctx context.Context, hostKey types.PublicKey, siamuxAddr string, timeout time.Duration) (pt rhpv3.HostPriceTable, err error) {
+	if timeout > 0 {
```

Comment:
I'd expect the timeout to be passed via `ctx` instead of deriving another one. Like we do for acquiring a lock on a contract.